gtk4-macros/src/lib.rs

```diff
@@ -42,18 +42,20 @@ use syn::{parse_macro_input, DeriveInput};
 ///
 ///     #[derive(Debug, Default, CompositeTemplate)]
 ///     #[template(file = "test/template.ui")]
-///     pub struct MyWidget {
+///     pub struct MyDialog {
```

Comment:
This should probably all be in a separate commit

Reply:
As the GTK developers don't want you to use the internal-child in your newly created custom widgets, I would  drop the example changes and add a section that explains the various inputs the `template` macro attribute can take.

Reply:
I meant the type renaming. That change makes sense but is independent from the other things in the commit :)